### ISSUE TITLE
Long term upstream fix ?

### ISSUE THREAD

**Issue by tikr7** [darinpp/yoga-slim-7#5]
Hi @darinpp 

thank you very much for posting the solution here! It took me a while to find this...

Do you know if somebody is working on that to fix it upstream? Is it a kernel thing? Who would be responsible for it?

Does it have something to do with this file in the end?
https://github.com/torvalds/linux/blob/master/sound/pci/hda/patch_realtek.c
`static void yoga7_14arb7_fixup_i2c(struct hda_codec *cdc,`

**Comment by insanie** [comment]
I don't believe this issue can be fixed in linux-firmware upstream since this exact firmware is tinkered specifically for this laptop. This means on another laptop model that uses TI 2781 it might (= most probably) sound aweful.
Selective (= based on laptop model) fw replacement seems to be the only way to go here. I don't see a clear way to address that in upstream.
